upstream-sha256: ff257f4321eb3fa2acf572934e509757c2202f4e0eba4eb4ab4a191533d059e9
actual-sha256: e2864f56df3fa3b5e5d1cc60550595a32e0f6e24f1ba4bdaf6b2b1898a394036
--- upstream.expected
+++ tolk.actual
@@ -17,8 +17,9 @@
       *(buf0+0) = ((*(buf0+0))+val0);
     }
     float val1 = (*(buf0+0));
-    float alu4 = ((val1<0.0f)?0.0f:val1);
-    *(buf1+0) = ((*(buf1+0))+alu4);
+    float alu4 = ((val1!=val1)?val1:0.0f);
+    float alu5 = ((0.0f<val1)?val1:alu4);
+    *(buf1+0) = ((*(buf1+0))+alu5);
   }
   *(buf3+lidx0) = (*(buf1+0));
   threadgroup_barrier(mem_flags::mem_threadgroup);
